src/Entities/Entity.cpp

```diff
@@ -242,10 +242,17 @@ void cEntity::TakeDamage(eDamageType a_DamageType, cEntity * a_Attacker, int a_R
 {
 	TakeDamageInfo TDI;
 	TDI.DamageType = a_DamageType;
-	TDI.Attacker = a_Attacker;
+	if ((a_Attacker != nullptr) && a_Attacker->IsPawn())
+	{
+		TDI.Attacker = a_Attacker;
+	}
+	else
+	{
+		TDI.Attacker = nullptr;
+	}
```

Comment:
@madmaxoft The proposed method produces code dupes, and forces TDI users to worry about things they shouldn't worry about. For instance, I'd need this in `cMonster.cpp::doTakeDamage`:

```
		if (a_TDI.Attacker->IsProjectile())
		{
			m_Target = nullptr;
			UInt32 AttackerId = static_cast<cProjectileEntity*>(a_TDI.Attacker)->GetCreatorUniqueID();

			class cCallback :
				public cEntityCallback
			{
			public:
				cPawn * m_PawnAttacker;
				virtual bool Item(cEntity * a_PawnAttacker) override
				{
					m_PawnAttacker = static_cast<cPawn*>(a_PawnAttacker);
					return m_PawnAttacker;
				}
			} Callback;
			Callback.m_PawnAttacker = nullptr;
			m_World->DoWithEntityByID(AttackerId, Callback);  // If the projectile shooter still exists, Callback.m_PawnAttacker will point to it
			m_Target = Callback.m_PawnAttacker;
		}
```

I'd need something similar in the player's `doTakeDamage` so that the player asks the wolves for help. And all plugins would need some similar piece of code.

If TDI.Attacker contained the actual attacker, the users of TDI wouldn't have to care about *how* the attacker dealt damage, unless they are actually interested in that.

I propse this:
```
TDI.PawnAttacker - The player/Mob who caused the damage. nullptr if no player or mob were involved. If this is projectile damage, PawnAttacker is the shooter, and Attacker is the projectile. If this is melee damage, PawnAttacker and Attacker are identical.
TDI.Attacker - The entity which dealt the damage. Might be a projectile, a mob, or a null. See TDI.PawnAttacker.
```